File: backend/src/test/java/github/lms/lemuel/xr/content/domain/TopicTest.java
package github.lms.lemuel.xr.content.domain;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.Test;

class TopicTest {

    @Test
    void byId_정합() {
        assertThat(Topic.byId(1)).isEqualTo(Topic.JOURNAL);
        assertThat(Topic.byId(7)).isEqualTo(Topic.FEAR);
    }

    @Test
    void 잘못된_id_예외() {
        assertThatThrownBy(() -> Topic.byId(99))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @Test
    void 모든_topic_은_id_와_제목_가짐() {
        for (Topic t : Topic.values()) {
            assertThat(t.id()).isBetween(1, 7);
            assertThat(t.title()).isNotBlank();
        }
    }
}

File: backend/src/test/kotlin/github/lms/lemuel/xr/content/domain/TopicTest.kt
package github.lms.lemuel.xr.content.domain

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class TopicTest {

    @Test
    fun `byId 정합`() {
        assertThat(Topic.byId(1)).isEqualTo(Topic.JOURNAL)
        assertThat(Topic.byId(7)).isEqualTo(Topic.FEAR)
    }

    @Test
    fun `잘못된 id 예외`() {
        assertThatThrownBy { Topic.byId(99) }
            .isInstanceOf(IllegalArgumentException::class.java)
    }

    @Test
    fun `모든 topic 은 id 와 제목 가짐`() {
        for (t in Topic.entries) {
            assertThat(t.id).isBetween(1, 7)
            assertThat(t.title).isNotBlank()
        }
    }
}
